### PR TITLE
Fixes for warnings produced by clang

### PR DESCRIPTION
**Pull request by bje-**

### CODE REVIEW

File: src/dview/dvplotctrl.cpp [hunk 1]
```diff
@@ -449,7 +449,7 @@ void wxDVPlotCtrl::SelectDataIndex(size_t index, bool allTabs)
 {
 	//Only select this data on the slower tabs (which require sorting) if allTabs is true.
 
-	if (index < 0 || index >= m_dataSets.size())
```
Comment (nickdiorio) [inline src/dview/dvplotctrl.cpp]: This is probably warning because a size_t cannot logically be < 0.

Comment (bje-) [reply src/dview/dvplotctrl.cpp]: That is exactly why it is warning.

File: src/pdf/pdffontdataopentype.cpp [hunk 2]
```diff
@@ -633,7 +633,6 @@ wxPdfFontDataOpenTypeUnicode::WriteUnicodeMap(wxOutputStream* mapData,
         if (subsetGlyphs != NULL)
         {
           glEntry->m_gid = (*subsetGlyphs)[charIter->second];
-          charIter->second;
```
Comment (nickdiorio) [inline src/pdf/pdffontdataopentype.cpp]: This returns the value pointed to by the iterator, but isn't used.

File: src/pdf/pdfgraphics.cpp [hunk 3]
```diff
@@ -1860,15 +1860,15 @@ wxPdfDocument::SetLineStyle(const wxPdfLineStyle& linestyle)
...
   {
-    wxString dashString = wxT("");
+    wxString dashString = wxS("");
     size_t j;
     for (j = 0; j < dash.GetCount(); j++)
     {
       if (j > 0)
       {
-        dashString += wxString(wxT(" "));
+        dashString += wxString(wxS(" "));
```
Comment (nickdiorio) [inline src/pdf/pdfgraphics.cpp]: Not clear to me what the differences in wxS and wxT according to the wxWidgets documentation, but this is fine.

Comment (bje-) [reply src/pdf/pdfgraphics.cpp]: I copied the function exactly from wxpdfdoc upstream and the upstream maintainer made that change. I decided to keep it to minimise divergence from the upstream code.

**Pull request merged**